lib/postgrex/utils.ex

```diff
@@ -47,15 +47,23 @@ defmodule Postgrex.Utils do
   Fills in the given `opts` with default options.
   """
   @spec default_opts(Keyword.t) :: Keyword.t
-  def default_opts(opts) do
+  def default_opts(opts, environment \\ System) do
     opts
-    |> Keyword.put_new(:username, System.get_env("PGUSER") || System.get_env("USER"))
-    |> Keyword.put_new(:password, System.get_env("PGPASSWORD"))
-    |> Keyword.put_new(:hostname, System.get_env("PGHOST") || "localhost")
-    |> Keyword.put_new(:port, System.get_env("PGPORT"))
+    |> Keyword.put_new(:username, environment.get_env("PGUSER") || environment.get_env("USER"))
+    |> Keyword.put_new(:password, environment.get_env("PGPASSWORD"))
+    |> Keyword.put_new(:hostname, environment.get_env("PGHOST") || "localhost")
+    |> Keyword.put_new(:port, set_port(environment.get_env("PGPORT")))
```

Comment:
You can call `System.put_env/2` in a test and call `System.delete_env/1` after instead of adding an optional argument.